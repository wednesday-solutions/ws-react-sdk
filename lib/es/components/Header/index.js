import{_ as e,b as n,a as t}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"antd/lib/dropdown/style";import o from"antd/lib/dropdown";import"antd/lib/badge/style";import r from"antd/lib/badge";import"antd/lib/layout/style";import i from"antd/lib/layout";import"../_rollup-plugin-inject-process-env-1259e43d.js";import a from"react";import l from"prop-types";import m from"styled-components";import{c as s}from"../colors-93a064d8.js";import{f as d}from"../fonts-ca6864a0.js";import{m as c}from"../media-ee9cbc51.js";import p from"../If/index.js";import g from"../T/index.js";import{injectIntl as h}from"react-intl";import{toUpper as f,get as u}from"lodash";import{BellOutlined as b,UserOutlined as x}from"@ant-design/icons";var y,w,v,E,N,j,k,z,I,O=m(i.Header)(y||(y=e(["\n  && {\n    &.ant-layout-header {\n      height: 6rem;\n      padding: 0 2rem;\n      border-bottom: 1px solid lightgray;\n      ",";\n      ",";\n    }\n    display: flex;\n    align-items: center;\n    background-color: ",";\n  }\n"])),c.tablet.max("\n         height: 5rem;\n         padding: 0 1rem;\n     "),c.largeMobile.max("\n       height: 3rem;\n     "),s.white),M=m.img(w||(w=e(["\n  height: 2.5rem;\n  width: 8.75rem;\n  display: flex;\n  align-items: center;\n  cursor: pointer;\n  ",";\n"])),c.largeMobile.max("\n   height: 1.475rem;\n   width: 5.1875rem;\n   ")),C=m.div(v||(v=e(["\n  flex: 1;\n  display: flex;\n  justify-content: flex-end;\n  align-items: center;\n  width: auto;\n"]))),L=m(b)(E||(E=e(["\n  ",";\n  ",";\n"])),d.size.large(),c.largeMobile.max("\n     ".concat(d.size.small(),";\n   "))),R=m(r)(N||(N=e(["\n  cursor: pointer;\n  .anticon {\n    color: ",";\n  }\n  .ant-badge-dot {\n    right: 0.4rem;\n    top: 0.1rem;\n  }\n"])),s.darkFuchsia),_=m.div(j||(j=e(["\n  display: flex;\n  position: relative;\n  height: 2.6rem;\n  width: 2.6rem;\n  margin-left: 1rem;\n  background: ",";\n  color: ",";\n  border-radius: 50%;\n  cursor: pointer;\n  ","\n  justify-content: center;\n  align-items: center;\n  ",";\n"])),s.primary,s.white,(e=>e.zIndex&&"z-index: 5;"),c.largeMobile.max("\n     height: 1.5rem;\n     width: 1.5rem\n   ")),q=m(g)(k||(k=e(["\n  ",";\n  position: absolute;\n  top: 50%;\n  left: 50%;\n  transform: translate(-50%, -50%);\n  ",";\n"])),d.size.big(),c.largeMobile.max("\n     ".concat(d.size.small(),";\n   "))),H=m.img(z||(z=e(["\n  width: 100%;\n  border-radius: 50px;\n"]))),P=m(x)(I||(I=e(["\n  && {\n    font-size: 1.5rem;\n  }\n"])));function T(e){var{logo:r,user:i,dropdownOverlay:l,handleOnLogoClick:m,hasNewNotifications:s,hasNotificationIcon:d}=e,c=n(e,["logo","user","dropdownOverlay","handleOnLogoClick","hasNewNotifications","hasNotificationIcon"]),g=f(u(i,"name","")),h=g?a.createElement("div",{"data-testid":"user-name-initial"},a.createElement(q,{text:g.charAt(0)})):a.createElement(P,null);return a.createElement(O,t({},c,{"data-testid":"header"}),a.createElement(M,{src:r,"data-testid":"logo",onClick:m}),a.createElement(C,null,a.createElement(p,{condition:d},a.createElement(R,{"data-testid":"notification-badge",dot:s},a.createElement(L,null))),a.createElement(o,{overlay:l,placement:"bottomRight",trigger:["hover"]},a.createElement(_,null,a.createElement(p,{condition:null==i?void 0:i.photoUrl,otherwise:h},a.createElement(H,{src:null==i?void 0:i.photoUrl}))))))}T.propTypes={logo:l.element.isRequired,user:l.object,dropdownOverlay:l.element,handleOnLogoClick:l.func.isRequired,hasNewNotifications:l.bool,hasNotificationIcon:l.bool},T.defaultProps={user:{name:""},hasNotificationIcon:!1,hasNewNotifications:!1};var U=h(T);export default U;export{P as DefaultImg,_ as ProfileContainer,H as ProfileImg,q as UserName};
