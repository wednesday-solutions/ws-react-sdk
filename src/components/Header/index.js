/**
 *
 * Header
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { Badge, Dropdown, Layout } from 'antd';
import styled from 'styled-components';
import { fonts, colors, media } from '@themes';
import If from '@components/If';
import T from '@components/T';
import injectIntl from '@components/injectIntl';
import { get, toUpper } from 'lodash';
import { UserOutlined, BellOutlined } from '@ant-design/icons';

const StyledHeader = styled(Layout.Header)`
  && {
    &.ant-layout-header {
      height: 6rem;
      padding: 0 2rem;
      border-bottom: 1px solid lightgray;
      ${media.tablet.max(`
         height: 5rem;
         padding: 0 1rem;
     `)};
      ${media.largeMobile.max(`
       height: 3rem;
     `)};
    }
    display: flex;
    align-items: center;
    background-color: ${colors.white};
  }
`;

const Logo = styled.img`
  height: 2.5rem;
  width: 8.75rem;
  display: flex;
  align-items: center;
  cursor: pointer;
  ${media.largeMobile.max(`
   height: 1.475rem;
   width: 5.1875rem;
   `)};
`;

const OptionContainer = styled.div`
  flex: 1;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  width: auto;
`;

const StyledBellFilled = styled(BellOutlined)`
  ${fonts.size.large()};
  ${media.largeMobile.max(`
     ${fonts.size.small()};
   `)};
`;
const StyledBadge = styled(Badge)`
  cursor: pointer;
  .anticon {
    color: ${colors.darkFuchsia};
  }
  .ant-badge-dot {
    right: 0.4rem;
    top: 0.1rem;
  }
`;

export const ProfileContainer = styled.div`
  display: flex;
  position: relative;
  height: 2.6rem;
  width: 2.6rem;
  margin-left: 1rem;
  background: ${colors.primary};
  color: ${colors.white};
  border-radius: 50%;
  cursor: pointer;
  ${props => props.zIndex && `z-index: 5;`}
  justify-content: center;
  align-items: center;
  ${media.largeMobile.max(`
     height: 1.5rem;
     width: 1.5rem
   `)};
`;

export const UserName = styled(T)`
  ${fonts.size.big()};
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
  ${media.largeMobile.max(`
     ${fonts.size.small()};
   `)};
`;
export const ProfileImg = styled.img`
  width: 100%;
  border-radius: 50px;
`;

export const DefaultImg = styled(UserOutlined)`
  && {
    font-size: 1.5rem;
  }
`;
const Title = styled(T)`
  && {
    margin-bottom: 0;
    ${fonts.dynamicFontSize(fonts.size.xRegular, 1, 0.5)};
    display: flex;
    align-self: center;
  }
`;

function Header({
  logo,
  user,
  dropdownOverlay,
  handleOnLogoClick,
  hasNewNotifications,
  hasNotificationIcon,
  header,
  ...props
}) {
  const userName = toUpper(get(user, 'name', ''));
  const defaultProfile = userName ? (
    <div data-testid="user-name-initial">
      <UserName text={userName.charAt(0)} />
    </div>
  ) : (
    <DefaultImg />
  );
  return (
    <StyledHeader {...props} data-testid="header">
      <Logo src={logo} data-testid="logo" onClick={handleOnLogoClick} />
      <If condition={header}>
        <Title data-testId='title' type="heading" id={header} />
      </If>
      <OptionContainer>
        <If condition={hasNotificationIcon}>
          <StyledBadge data-testid="notification-badge" dot={hasNewNotifications}>
            <StyledBellFilled />
          </StyledBadge>
        </If>
        <If condition={dropdownOverlay}>
          <Dropdown overlay={dropdownOverlay} placement="bottomRight" trigger={['hover']}>
            <ProfileContainer >
              <If condition={user?.photoUrl} otherwise={defaultProfile}>
                <ProfileImg src={user?.photoUrl} />
              </If>
            </ProfileContainer>
          </Dropdown>
        </If>
      </OptionContainer>
    </StyledHeader>
  );
}

Header.propTypes = {
  logo: PropTypes.element.isRequired,
  user: PropTypes.object,
  dropdownOverlay: PropTypes.element,
  handleOnLogoClick: PropTypes.func.isRequired,
  hasNewNotifications: PropTypes.bool,
  hasNotificationIcon: PropTypes.bool,
  header: PropTypes.string
};

Header.defaultProps = {
  user: {
    name: ''
  },
  hasNotificationIcon: false,
  hasNewNotifications: false
};

export default injectIntl(Header);
