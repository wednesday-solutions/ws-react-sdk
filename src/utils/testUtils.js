import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
