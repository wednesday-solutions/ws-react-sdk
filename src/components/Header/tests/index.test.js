/**
 *
 * Tests for Header
 *
 */

import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import Header from '../index';

describe('<Header />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<Header />);
    expect(baseElement).toMatchSnapshot();
  });
});
