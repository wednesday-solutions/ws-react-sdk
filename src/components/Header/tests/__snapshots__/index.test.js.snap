// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<p
  class="sc-bdnxRM heVfDC sc-jrsJWt hdOySf"
  data-testid="title"
>
  wednesday
</p>
`;

exports[` 2`] = `
<span
  class="ant-badge sc-jSFjdj bBYqLn"
  data-testid="notification-badge"
>
  <span
    aria-label="bell"
    class="anticon anticon-bell sc-eCApnc cFEzwn"
    role="img"
  >
    <svg
      aria-hidden="true"
      data-icon="bell"
      fill="currentColor"
      focusable="false"
      height="1em"
      viewBox="64 64 896 896"
      width="1em"
    >
      <path
        d="M816 768h-24V428c0-141.1-104.3-257.7-240-277.1V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.9c-135.7 19.4-240 136-240 277.1v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48zM304 768V428c0-55.6 21.6-107.8 60.9-147.1S456.4 220 512 220c55.6 0 107.8 21.6 147.1 60.9S720 372.4 720 428v340H304z"
      />
    </svg>
  </span>
</span>
`;

exports[`<Header /> should render and match the snapshot 1`] = `
<body>
  <div>
    <header
      class="ant-layout-header sc-gsTCUz fNPrnw"
      data-testid="header"
      intl="[object Object]"
    >
      <img
        class="sc-dlfnbm fruiyx"
        data-testid="logo"
      />
      <p
        class="sc-bdnxRM heVfDC sc-jrsJWt hdOySf"
        data-testid="title"
      >
        wednesday
      </p>
      <div
        class="sc-hKFxyN cyCtHb"
      />
    </header>
  </div>
  <div>
    <header
      class="ant-layout-header sc-gtsrHT eKqqMW"
      data-testid="header"
      intl="[object Object]"
    >
      <img
        class="sc-dlnjwi UbOhE"
        data-testid="logo"
      />
      <div
        class="sc-hKFxyN cyCtHb"
      >
        <span
          class="ant-badge sc-jSFjdj bBYqLn"
          data-testid="notification-badge"
        >
          <span
            aria-label="bell"
            class="anticon anticon-bell sc-eCApnc cFEzwn"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="bell"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M816 768h-24V428c0-141.1-104.3-257.7-240-277.1V112c0-22.1-17.9-40-40-40s-40 17.9-40 40v38.9c-135.7 19.4-240 136-240 277.1v340h-24c-17.7 0-32 14.3-32 32v32c0 4.4 3.6 8 8 8h216c0 61.8 50.2 112 112 112s112-50.2 112-112h216c4.4 0 8-3.6 8-8v-32c0-17.7-14.3-32-32-32zM512 888c-26.5 0-48-21.5-48-48h96c0 26.5-21.5 48-48 48zM304 768V428c0-55.6 21.6-107.8 60.9-147.1S456.4 220 512 220c55.6 0 107.8 21.6 147.1 60.9S720 372.4 720 428v340H304z"
              />
            </svg>
          </span>
        </span>
      </div>
    </header>
  </div>
  <div>
    <header
      class="ant-layout-header sc-gtsrHT eKqqMW"
      data-testid="header"
      intl="[object Object]"
    >
      <img
        class="sc-dlnjwi UbOhE"
        data-testid="logo"
      />
      <div
        class="sc-hKFxyN cyCtHb"
      />
    </header>
  </div>
</body>
`;
